src/main/java/org/opentripplanner/routing/api/request/refactor/request/NewRouteRequest.java

```diff
@@ -0,0 +1,470 @@
+        }
+        case CAR -> journeyRequest.setStreetSubRequestModes(new TraverseModeSet(TraverseMode.CAR));
+        case CAR_TO_PARK -> {
+          journeyRequest.setStreetSubRequestModes(
+            new TraverseModeSet(TraverseMode.CAR, TraverseMode.WALK)
+          );
+          routingPreferences.car().setParkAndRide(true);
+        }
+        case CAR_PICKUP -> {
+          journeyRequest.setStreetSubRequestModes(
+            new TraverseModeSet(TraverseMode.CAR, TraverseMode.WALK)
+          );
+          routingPreferences.car().allowPickup();
+        }
+        case CAR_RENTAL -> {
+          journeyRequest.setStreetSubRequestModes(
+            new TraverseModeSet(TraverseMode.CAR, TraverseMode.WALK)
+          );
+          routingPreferences.rental().setAllow(true);
+          journeyRequest.rental().allowedFormFactors().add(RentalVehicleType.FormFactor.CAR);
+        }
+      }
+    }
+
+    return Pair.of(streetRequest, streetPreferences);
+  }
+
+  /**
+   * Returns if the vertex is considered "close" to the start or end point of the request. This is
+   * useful if you want to allow loops in car routes under certain conditions.
+   * <p>
+   * Note: If you are doing Raptor access/egress searches this method does not take the possible
+   * intermediate points (stations) into account. This means that stations might be skipped because
+   * a car route to it cannot be found and a suboptimal route to another station is returned
+   * instead.
+   * <p>
+   * If you encounter a case of this, you can adjust this code to take this into account.
+   *
+   * @see NewRouteRequest#MAX_CLOSENESS_METERS
+   * @see DominanceFunction#betterOrEqualAndComparable(State, State)
+   */
+  public boolean isCloseToStartOrEnd(Vertex vertex) {
+    if (from == null || to == null || from.getCoordinate() == null || to.getCoordinate() == null) {
+      return false;
+    }
+    if (fromEnvelope == null) {
+      fromEnvelope = getEnvelope(from.getCoordinate(), MAX_CLOSENESS_METERS);
+    }
+    if (toEnvelope == null) {
+      toEnvelope = getEnvelope(to.getCoordinate(), MAX_CLOSENESS_METERS);
+    }
+    return (
+      fromEnvelope.intersects(vertex.getCoordinate()) ||
+      toEnvelope.intersects(vertex.getCoordinate())
+    );
+  }
+
+  private static Envelope getEnvelope(Coordinate c, int meters) {
+    double lat = SphericalDistanceLibrary.metersToDegrees(meters);
+    double lon = SphericalDistanceLibrary.metersToLonDegrees(meters, c.y);
+
+    Envelope env = new Envelope(c);
+    env.expandBy(lon, lat);
+
+    return env;
+  }
+
+  public NewRouteRequest reversedClone() {
+    var request = this.clone();
+    request.setArriveBy(!request.arriveBy);
+
+    return request;
+  }
+
+  private void setJourney(JourneyRequest journey) {
+    this.journey = journey;
+  }
+
+  public NewRouteRequest clone() {
+    try {
+      // TODO: 2022-08-25 there are some fields which will not be cloned in proper way
```

Comment:
Isn't it a deep clone right now? I now there are some fields that are not cloned but that how it was before the refactor so I did not touch them. But otherwise it should be a clone.